workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,238 @@
+* Copy the code as you go through the topics below.
+
+
+In the following example, you choose to use [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html#what-is-a-parameter) to centrally store a configuration information, such as the username for your database. For this, you choose to describe an `AWS::SSM::Parameter` resource in your CloudFormation template where you will store the username value. You then choose to write an IAM policy to describe action(s) you wish to grant to an AWS [Lambda function](https://aws.amazon.com/lambda/) you will create and use to consume your parameter store value. You choose to start with describing your IAM policy, where you reference the SSM parameter you create: this will require you to know the Amazon Resource Name (ARN) of the SSM parameter. First, you check the return values section for the resource (the SSM parameter in this case): in the relevant [documentation page](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-parameter.html#aws-resource-ssm-parameter-return-values), you see that `Ref` returns the parameter name, and `Fn::GetAtt` returns the type and the value. Since the ARN is not available as an output value for that resource type today, you can choose to leverage pseudo parameters to construct the ARN.
+
+
+
+You want to construct the resource ARN using pseudo parameters like the one in the following example. Let's say, for example, you have a sample parameter called `dbUsername` that you create in the `us-east-1` region, and in the AWS account `111122223333`; an example parameter ARN is constructed as in this example policy snippet:
+
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:GetParameters"
+            ],
+            "Resource": "arn:aws:ssm:us-east-1:111122223333:parameter/dbUsername"
+        }
+    ]
+}
+```
+Let's describe resources you need in your CloudFormation template.
+
+First, let’s start by defining your SSM parameter in the template: you will use a simple example of an SSM parameter where you define a parameter `Name` called `dbUsername`, and a `Value` you set to `alice`. Choose to copy content shown next, and paste it in the `pseudo-parameters.yaml` file, by appending it to the existing file content:
+
+```YAML
+Resources:
+  BasicParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: dbUsername
+      Type: String
+      Value: alice
+      Description: SSM Parameter for database username.
+```
+
+Next, let’s define the IAM role and policy from where you want to reference the SSM parameter you defined above. Again, choose to copy content shown next, and paste it in the Resources section of the `pseudo-parameters.yaml` file by appending it to the existing file content:
+
+```yaml
+  DemoRole:
+    Type: 'AWS::IAM::Role'
```

Comment:
Please remove quotes for the type here and in your template files for this lab, for consistency.